kornia/augmentation/random_generator/_2d/gaussian_illumination.py

```diff
@@ -0,0 +1,121 @@
+    """
+
+    def __init__(
+        self,
+        gain: tuple[float, float],
+        center: tuple[float, float],
+        sigma: tuple[float, float],
+        sign: tuple[float, float],
+    ) -> None:
+        super().__init__()
+        self.gain = gain
+        self.center = center
+        self.sigma = sigma
+        self.sign = sign
+
+    def __repr__(self) -> str:
+        r"""Return a string representation of the object."""
+        repr = f"gain={self.gain}, center={self.center}, sigma={self.sigma}, sign={self.sign}"
+        return repr
+
+    def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
+        r"""Create samplers for generating random gaussian illumination parameters."""
+
+        gain = _range_bound(
+            self.gain,
+            "gain",
+        ).to(device, dtype)
+        self.gain_sampler = UniformDistribution(gain[0], gain[1], validate_args=False)
+
+        center = _range_bound(
+            self.center,
+            "center",
+        ).to(device, dtype)
+        self.center_sampler = UniformDistribution(center[0], center[1], validate_args=False)
+
+        sigma = _range_bound(
+            self.sigma,
+            "sigma",
+        ).to(device, dtype)
+        self.sigma_sampler = UniformDistribution(sigma[0], sigma[1], validate_args=False)
+
+        sign = _range_bound(self.sign, "sign", bounds=(-1.0, 1.0), center=0.0).to(device, dtype)
+        self.sign_sampler = UniformDistribution(sign[0], sign[1], validate_args=False)
+
+    def forward(self, batch_shape: tuple[int, ...], same_on_batch: bool = False) -> dict[str, Tensor]:
+        r"""Generate random 2D Gaussian illumination patterns."""
+        batch_size, channels, height, width = batch_shape
+        _common_param_check(batch_size, same_on_batch)
+        _device, _dtype = _extract_device_dtype([self.gain, self.center, self.sigma])
+
+        gain_factor = _adapted_rsampling((batch_size,), self.gain_sampler, same_on_batch).to(
+            device=_device, dtype=_dtype
+        )
+        sigma_x = width * _adapted_rsampling((batch_size,), self.sigma_sampler, same_on_batch).to(
+            device=_device, dtype=_dtype
+        )
+        center_x = torch.round(width * _adapted_rsampling((batch_size,), self.center_sampler, same_on_batch)).to(
+            device=_device, dtype=_dtype
+        )
+        sigma_y = height * _adapted_rsampling((batch_size,), self.sigma_sampler, same_on_batch).to(
+            device=_device, dtype=_dtype
+        )
+        center_y = torch.round(height * _adapted_rsampling((batch_size,), self.center_sampler, same_on_batch)).to(
+            device=_device, dtype=_dtype
+        )
+        sign = torch.where(
+            _adapted_rsampling((batch_size,), self.sign_sampler, same_on_batch) >= 0.0,
+            torch.tensor(1),
+            torch.tensor(-1),
+        ).to(device=_device, dtype=_dtype)
+
+        gradient = torch.zeros(batch_shape)
+        for batch in range(batch_size):
```

Comment:
why it cannot be done in batch ?

Reply:
I'm not sure, but in order to do it in batch I have to be able to generate the gaussians in batch and I didn't know how to do it. If you can help me, we will integrate it.